corehq/apps/users/bulkupload.py

```diff
@@ -545,7 +546,7 @@ def _set_progress(progress):
                         # Without this line, digest auth doesn't work.
                         # With this line, digest auth works.
                         # Other than that, I'm not sure what's going on
-                        user.get_django_user(db_name=get_default_db_aliases()[0]).check_password(password)
+                        user.get_django_user(db_name=db_for_read_write(User, write=True)).check_password(password)
```

Comment:
I still think having a general utility to do this would be handy:
```
with force_db_read_from_primary(db_name):
    # do something
```

Reply:
Thoughts on merging as-is to fix the time sensitive bug and afterwards I can add the context manager, write tests, etc?

Reply:
yea, not blocking at all - probably not worth another PR now unless we find a second use case for it.